deeplake/integrations/mmdet/mmdet_runners.py

```diff
@@ -0,0 +1,116 @@
+import mmcv
+import torch
+import logging
+from mmcv import runner
+from torch.utils.data import DataLoader
+
+import time
+import warnings
+from typing import List, Tuple, Optional
+
+
+TIME_INTERVAL_FOR_CUDA_MEMORY_CLEANING = 10 * 60
+
+
+@runner.RUNNERS.register_module()
+class DeeplakeIterBasedRunner(runner.IterBasedRunner):
+    def run(
+        self,
+        data_loaders: List[DataLoader],
+        workflow: List[Tuple[str, int]],
+        max_iters: Optional[int] = None,
+        **kwargs,
+    ) -> None:
+        """Start running.
+
+        Args:
+            data_loaders (list[:obj:`DataLoader`]): Dataloaders for training
+                and validation.
+            workflow (list[tuple]): A list of (phase, iters) to specify the
+                running order and iterations. E.g, [('train', 10000),
+                ('val', 1000)] means running 10000 iterations for training and
+                1000 iterations for validation, iteratively.
+        """
+        assert isinstance(data_loaders, list)
+        assert mmcv.is_list_of(workflow, tuple)
+        assert len(data_loaders) == len(workflow)
+        if max_iters is not None:
+            warnings.warn(
+                "setting max_iters in run is deprecated, "
+                "please set max_iters in runner_config",
+                DeprecationWarning,
+            )
+            self._max_iters = max_iters
+        assert (
+            self._max_iters is not None
+        ), "max_iters must be specified during instantiation"
+
+        work_dir = self.work_dir if self.work_dir is not None else "NONE"
+        self.logger.info(
+            "Start running, host: %s, work_dir: %s",
+            runner.utils.get_host_info(),
+            work_dir,
+        )
+        self.logger.info(
+            "Hooks will be executed in the following order:\n%s", self.get_hook_info()
+        )
+        self.logger.info("workflow: %s, max: %d iters", workflow, self._max_iters)
+        self.call_hook("before_run")
+
+        iter_loaders = [runner.IterLoader(x) for x in data_loaders]
+
+        self.call_hook("before_epoch")
+
+        formatter = logging.Formatter("%(relative)ss")
+        start_time = time.time()
```

Comment:
all of the code is taken from mmcv.runner.iter_based_runner.py, the only difference is that I added time.time